relay-server/src/services/processor.rs

```diff
@@ -2967,21 +2962,22 @@ impl UpstreamRequest for SendMetricsRequest {
                     response.consume().await.ok();
                 }
                 Err(error) => {
+                    relay_log::error!(error = &error as &dyn Error, "Failed to send metrics batch");
+
                     // If the request did not arrive at the upstream, we are responsible for outcomes.
                     // Otherwise, the upstream is responsible to log outcomes.
-                    if !error.is_received() {
-                        for (scoping, quantities) in self.quantities {
-                            utils::reject_metrics::<&Bucket>(
-                                &self.outcome_aggregator,
-                                &self.metric_stats,
-                                quantities,
-                                scoping,
-                                Outcome::Invalid(DiscardReason::Internal),
-                                None,
-                            );
-                        }
+                    if error.is_received() {
+                        return;
+                    }
+
+                    for (scoping, quantities) in self.quantities {
+                        self.metric_outcomes.track(
+                            scoping,
+                            &[] as &[Bucket], // TODO(dav1d): provide buckets here
```

Comment:
Is this for a follow-up?

Reply:
Yes! We had the same problem before, it was the `None` in the `reject_metrics` call. I decided not to also add these functional changes to the PR, will do in a follow up.